frontend/packages/git-service/src/utils/pac-strategy-detector.ts

```diff
@@ -0,0 +1,50 @@
+import { BaseService } from '../services/base-service';
+
+type PacData = {
+  isPipelinePresent: boolean;
+  pipelineFiles: string[];
+};
+
+export const pacfilesDetection = async (gitService: BaseService) => {
+  let pacFiles = [];
+  const isTektonFolderPresent = await gitService.isTektonFolderPresent();
+
+  if (isTektonFolderPresent) {
+    pacFiles.push('.tekton');
+    const { files: pipelineFiles } = await gitService.getRepoFileList('.tekton');
+    if (pipelineFiles.length > 0) {
+      pacFiles = [...pacFiles, ...pipelineFiles];
+    }
+  }
+  return pacFiles;
+};
+
+export const verifyPipelinesFile = async (file: string, gitService: BaseService) => {
+  const resourceContent = await gitService.getFileContent(`.tekton/${file}`);
+  if (resourceContent.includes('kind: PipelineRun')) {
+    return true;
+  }
+  return false;
+};
+
+export const isPipelineFilesPresent = (pacFiles: string[], gitService: BaseService): PacData => {
+  if (!pacFiles || pacFiles?.length === 0) {
+    return {
+      isPipelinePresent: false,
+      pipelineFiles: [],
+    };
+  }
+  let pipelineFiles: string[] = pacFiles.filter(
+    (file) => file?.endsWith('.yaml') || file?.endsWith('.yml'),
```

Comment:
```suggestion
    (file) => (file?.endsWith('.yaml') || file?.endsWith('.yml')) && verifyPipelinesFile(file, gitService)),
```

Reply:
Also instead of `filter` you can use `find` since at the moment we just need one PLR file to be present.